crates/rust-analyzer/src/config.rs

```diff
@@ -7,12 +7,40 @@
 //! configure the server itself, feature flags are passed into analysis, and
 //! tweak things like automatic insertion of `()` in completions.
 
-use ra_project_model::InlayHintOptions;
+use ra_ide::{InlayConfig, InlayKind};
 use rustc_hash::FxHashMap;
 
 use ra_project_model::CargoFeatures;
 use serde::{Deserialize, Deserializer};
 
+#[derive(Deserialize)]
+#[serde(remote = "InlayKind")]
+pub enum InlayKindDef {
```

Comment:
I think there's already an enum for this entity:
https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/rust-analyzer/src/req.rs#L199

so let's not create a copy of it.
A few more observations, feel free to implement any (or none) of them, if you want to:

* It could be a bit simpler to use a `match` to map between both `InlayKind` enums, similar to the https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/rust-analyzer/src/main_loop/handlers.rs#L1006

* Since you can easily deserialize a `Vec<InlayKind>`, you can ask for that in the `package.json` instead and get rid of all changes in the `ra_project_model` crate.
Less code with more flexibility for the end users (as long as they know which options are available)

Reply:
Does it make sense to deduplicate the `InlayKind` as mentioned and then just use the serde remote feature so we don't even need to do the match in the first place?

Reply:
I sent a diff with this change so you can see what I mean

Reply:
Personally, I'd prefer using `match`, but if your way works, it's also good, thanks for the dedup.